lib/Rex/Commands/Iptables.pm

```diff
@@ -548,4 +610,25 @@ sub _rule_exists {
   return 0;
 }
 
+sub _get_ip_version {
+  my ($params) = @_;
+  if ( defined $params->[0] && !ref $params->[0] ) {
+    if ( $params->[0] eq "-4" || $params->[0] eq "-6" ) {
+      return shift @$params;
+    }
+  }
+  return -4;
+}
+
+sub _get_executable {
+  my ($params) = @_;
+  return _get_ip_version($params) == -6 ? "ip6tables" : "iptables";
+}
+
+sub _iptables_version {
+  my $version = join "", map { sprintf "%03d", $_ } split /[.]/,
+    substr run("iptables -V"), 10;
+  return $version =~ s/^0+//r;
```

Comment:
This syntax looks like introduced in perl-5.13.2. Can you modify it to run on perl-5.8.9?

You can find more info on perl compatibility goals on #578. You can also try to catch those errors early by running `dzil test --no-author --release` on your machine (author tests are being fixed, but right now you can skip perlcritic warnings woth `--no-author`).